packages/trpc/server/routers/publicViewer/noShow.handler.ts

```diff
@@ -31,33 +53,73 @@ export const noShowHandler = async ({ input }: NoShowOptions) => {
           email: true,
         },
       });
-
+      const allAttendeesMap = allAttendees.reduce((acc, attendee) => {
+        acc[attendee.email] = attendee;
+        return acc;
+      }, {} as Record<string, { id: number; email: string }>);
       const updatePromises = attendees.map((attendee) => {
-        const attendeeToUpdate = allAttendees.find((a) => a.email === attendee.email);
-
-        if (attendeeToUpdate) {
-          return prisma.attendee.update({
-            where: { id: attendeeToUpdate.id },
-            data: { noShow: attendee.noShow },
-          });
-        }
+        const attendeeToUpdate = allAttendeesMap[attendee.email];
+        if (!attendeeToUpdate) return;
+        return prisma.attendee.update({
+          where: { id: attendeeToUpdate.id },
+          data: { noShow: attendee.noShow },
+        });
       });
-
-      await Promise.all(updatePromises);
-    } else {
-      await prisma.booking.update({
-        where: {
-          uid: bookingUid,
-        },
-        data: {
-          noShowHost: true,
+      const results = await Promise.allSettled(updatePromises);
+      logFailedResults(results);
+      const _attendees = results
+        .filter((x) => x.status === "fulfilled")
+        .map((x) => (x as PromiseFulfilledResult<{ noShow: boolean; email: string }>).value)
+        .map((x) => ({ email: x.email, noShow: x.noShow }));
+      const payload = await getResultPayload(_attendees);
+      const booking = await prisma.booking.findUnique({
+        where: { uid: bookingUid },
+        select: {
+          eventType: {
+            select: {
+              id: true,
+              teamId: true,
+              userId: true,
+            },
+          },
         },
       });
+      const orgId = await getOrgIdFromMemberOrTeamId({
+        memberId: booking?.eventType?.userId,
+        teamId: booking?.eventType?.teamId,
+      });
+      const webhooks = await new WebhookService({
+        teamId: booking?.eventType?.teamId,
+        userId: booking?.eventType?.userId,
+        eventTypeId: booking?.eventType?.id,
+        orgId,
+        triggerEvent: WebhookTriggerEvents.BOOKING_NO_SHOW_UPDATED,
+      });
+
+      const t = await getTranslation("en", "common");
+      await webhooks.sendPayload({
+        ...payload,
+        /** We send webhook message pre-translated, on client we already handle this */
+        // @ts-expect-error payload is too booking specific, we need to refactor this
+        message: t(payload.message, { x: payload.attendees[0]?.email || "User" }),
+        bookingUid,
+      });
+      return payload;
     }
+    await prisma.booking.update({
+      where: {
+        uid: bookingUid,
+      },
+      data: {
+        noShowHost: true,
+      },
+    });
```

Comment:
This was inside else clause here https://github.com/calcom/cal.com/blob/main/packages/trpc/server/routers/publicViewer/noShow.handler.ts#L47.